demo/cmd/common/books.go

```diff
@@ -102,10 +103,13 @@ func GetBooks(participantName string, meshExpectedResponseCode int, egressExpect
 
 	// The URLs this participant will attempt to query from the bookstore service
 	urlSuccessMap := map[string]bool{
-		booksBought:    false,
-		buyBook:        false,
-		httpEgressURL:  false,
-		httpsEgressURL: false,
+		booksBought: false,
+		buyBook:     false,
+	}
+
+	if enableEgress {
+		urlSuccessMap[httpEgressURL] = false
+		urlSuccessMap[httpsEgressURL] = false
```

Comment:
not going to request a change, but since you parametrized the urlSuccessMap I'd have said to parametrized urlExpectedRespCode too

Reply:
It shouldn't matter since response code checks for egress only apply if the egress URLs aren't going to be used. I found it cleaner to avoid if/else unless necessary. The purpose of the change is not to wrap every egress related code in conditionals, but rather avoid egress requests unless enabled. WDYT?

Reply:
i'm fine with that